File: dotnet/src/MyDotey.SCF/Type/TypeConverter.cs
using System;
using sType = System.Type;

namespace MyDotey.SCF.Type
{
    /**
     * @author koqizhao
     *
     * May 21, 2018
     * 
     * convert a value from source type to target type
     */
    public interface TypeConverter<S, T> : TypeConverter
    {
        T Convert(S source);
    }

    public interface TypeConverter
    {
        sType SourceType { get; }
        sType TargetType { get; }

        object Convert(object source);
    }

}

File: java/src/main/java/org/mydotey/scf/type/TypeConverter.java
package org.mydotey.scf.type;

/**
 * @author koqizhao
 *
 * May 21, 2018
 * 
 * convert a value from source type to target type
 */
public interface TypeConverter<S, T> {

    Class<S> getSourceType();

    Class<T> getTargetType();

    T convert(S source);

}
